### PR TITLE
Add option to app host installer whether or not ot add a main

### PR DESCRIPTION
**Pull request by dostrander**
- App specs should not add main

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Add option to app host installer whether or not ot add a main  
  [dostrander](https://github.com/dostrander)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dostrander** [comment]
@segiddins @dnkoutso

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,10 +8,19 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
...
+* Don't add main for app specs.  
+  [Derek Ostrander](https://github.com/dostrander)
+  [#8235](https://github.com/CocoaPods/CocoaPods/pull/8235)
+
 * Add documentation for the Podfile installation options  
   [Eric Amorde](https://github.com/amorde)
   [#8198](https://github.com/CocoaPods/CocoaPods/issues/8198)
   [guides.cocoapods.org #142](https://github.com/CocoaPods/guides.cocoapods.org/issues/142)
+  
+* Adds app spec project integration.
```
Comment (dnkoutso) [inline CHANGELOG.md]: ha I think I already took care of that in another PR

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb [hunk 2]
```diff
@@ -33,6 +33,10 @@ class AppHostInstaller
           #
           attr_reader :app_target_label
 
+          # @return [Boolean] whether the app host installer should add main
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: should add a `main.m` file

File: CHANGELOG.md (hunk 1 above)
Comment (dostrander) [reply CHANGELOG.md]: 👍

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb [hunk 3]
```diff
@@ -41,14 +45,16 @@ class AppHostInstaller
           # @param [String] subgroup_name @see #subgroup_name
           # @param [String] group_name @see #group_name
           # @param [String] app_target_label see #app_target_label
+          # @param [Boolean] add_main see #add_main
           #
-          def initialize(sandbox, project, platform, subgroup_name, group_name, app_target_label)
+          def initialize(sandbox, project, platform, subgroup_name, group_name, app_target_label, add_main = true)
```
Comment (segiddins) [inline lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: nit: `add_main_file: true`

File: CHANGELOG.md (hunk 1 above)
Comment (dnkoutso) [reply CHANGELOG.md]: you do not need this change at all.

**Pull request merged**